core/src/main/java/org/owasp/dependencycheck/analyzer/AbstractSuppressionAnalyzer.java

```diff
@@ -113,9 +122,24 @@ protected void analyzeDependency(Dependency dependency, Engine engine) throws An
         if (rules.isEmpty()) {
             return;
         }
-        rules.forEach((rule) -> rule.process(dependency));
+        int ctr = 0;
+        for (SuppressionRule rule : rules.list()) {
+            if (filter(rule)) {
+                ctr++;
+                rule.process(dependency);
+            }
+        }
     }
 
+    /**
+     * Determines whether or not to filter a suppression rule.
+     *
+     * @param rule the suppression rule to evaluate
+     * @return <code>true</code> if the rule should be filtered; otherwise
+     * <code>true</code>
```

Comment:
@return documented to always retain true. And I think the description can be improved to be more clear on whether it indicates 'filtering out' or 'filtering in' of rules.
```suggestion
     * Determines whether a suppression rule should be retained when filtering a set of suppression rules for a concrete suppression analyzer.
     *
     * @param rule the suppression rule to evaluate
     * @return <code>true</code> if the rule should be retained; otherwise
     * <code>false<code>
```